index.js

```diff
@@ -430,6 +427,82 @@ instance.prototype.updateScenesAndSources = async function() {
 		}
 	}
 
+	let updateSceneSources = (source, scene) => {
+		if (self.sources[source.name] && self.sources[source.name]['visible']) {
+			self.sources[source.name]['visible'] = true;
+		} else {
+			self.sources[source.name] = source;
+			if (source.name == sceneList.currentScene) {
+				self.sources[source.name]['visible'] = true;
+			}
+			if (source.render === true && scene.name == sceneList.currentScene) {
+				self.sources[source.name]['visible'] = true;
+				for (let nestedSource of self.scenes[source.name].sources) {
+					self.sources[nestedSource.name] = nestedSource;
+					if (nestedSource.render === true && nestedSource.type == 'scene') {
+						self.sources[nestedSource.name]['visible'] = true;
+						for (let nestedSceneSource of self.scenes[nestedSource.name].sources) {
+							self.sources[nestedSceneSource.name] = nestedSceneSource;
+							if (nestedSceneSource.render === true) {
+								self.sources[nestedSceneSource.name]['visible'] = true;
+							}
+							if (nestedSceneSource.render === true && nestedSceneSource.type === 'group') {
+								updateGroupedSources(nestedSceneSource, scene)
+							}
+						}
+					} else if (nestedSource.render === true && nestedSource.type == 'group') {
+						self.sources[nestedSource.name]['visible'] = true;
+						updateGroupedSources(nestedSource, scene)
+					} else if (nestedSource.render === true) {
+						self.sources[nestedSource.name]['visible'] = true;
+					}
+				}
+			}
+		}
+	}
+
+	let updateGroupedSources = (source, scene) => {
+		if (source.render === true && self.sources[source.name] && scene.name == sceneList.currentScene) {
+			self.sources[source.name]['visible'] = true;
+		}
+		if (source.render === true) {
+			for (var s in source.groupChildren) {
```

Comment:
You're mixing `for (var s in ...)` with  `for (let s in ...)` throughout this PR.

Reply:
Changed this to be consistent for all